### PR TITLE
Introduce Ballerina Guidewire insurance-now connector

### LINKED ISSUE #6044: Write OpenAPI specification for Guidewire InsuranceNow Cloud API
**Description:**
> $subject

### LINKED ISSUE #6214: Introduce Ballerina Guidewire InsuranceNow connector
**Description:**
> $subject

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#6044](https://github.com/ballerina-platform/ballerina-library/issues/6044)
Fixes [#6214](https://github.com/ballerina-platform/ballerina-library/issues/6214)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: docs/spec/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,3026 @@
...
+              - desc
+            default: asc
+          description: Indicates the method used to sort the results by name.
+      responses:
+        "200":
+          description: Successful response.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ListCountryResponse"
```
Comment (lnash94) [inline docs/spec/openapi.yml]: Shall we rename this object schema for `ListCountry` or something without `Response`, having `Response` in the name can be misleading to a section with Response in the component response schema. 
Please check other Response schemas.

Comment (ayeshLK) [reply docs/spec/openapi.yml]: Fixed with [4f822b353b17aa352032637e305d3dbfddc796ab](https://github.com/ballerina-platform/module-ballerinax-guidewire.insnow/pull/2/commits/4f822b353b17aa352032637e305d3dbfddc796ab)

Comment (NipunaRanasinghe) [reply docs/spec/openapi.yml]: @lnash94 Just out of curiosity, what if we have a request schema as well for the same operation? :) don't we have have to use `[Foo]Request` and `[Foo]Response` prefixes in such scenarios anyway?

**Review by NipunaRanasinghe — DISMISSED** [review]
LGTM

File: docs/spec/openapi.yml [hunk 2]
```diff
@@ -0,0 +1,3602 @@
+openapi: "3.0.2"
+info:
+  title: GuideWire insuranceNow cloud API
+  version: 5.0.0
```
Comment (NipunaRanasinghe) [inline docs/spec/openapi.yml:4]: Here we better add a high-level description, which is a common practice in well-written openAPI contracts

Comment (ayeshLK) [reply docs/spec/openapi.yml:4]: Added a description with [27d693300fa4e0c1efec5f6911c0d1603a5a0317](https://github.com/ballerina-platform/module-ballerinax-guidewire.insnow/pull/2/commits/27d693300fa4e0c1efec5f6911c0d1603a5a0317)

**Review by NipunaRanasinghe — DISMISSED** [review]
LGTM

File: ballerina/utils.bal [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    boolean explode = true;
+    # Specifies the custom content type
+    string contentType?;
+    # Specifies the custom headers
+    map<any> headers?;
+};
+
+enum EncodingStyle {
+    DEEPOBJECT, FORM, SPACEDELIMITED, PIPEDELIMITED
+}
```
Comment (Nuvindu) [inline ballerina/utils.bal:38]:
```suggestion
enum EncodingStyle {
    DEEPOBJECT,
    FORM,
    SPACEDELIMITED,
    PIPEDELIMITED
}
```

File: ballerina/utils.bal [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+#
+# + parent - Parent record name
+# + anyRecord - Record to be serialized
+# + explode - Specifies whether arrays and objects should generate separate parameters
+# + return - Serialized record as a string
+isolated function getFormStyleRequest(string parent, record {} anyRecord, boolean explode = true) returns string {
+    string[] recordArray = [];
+    if explode {
+        foreach [string, anydata] [key, value] in anyRecord.entries() {
+            if (value is SimpleBasicType) {
```
Comment (Nuvindu) [inline ballerina/utils.bal:77]:
```suggestion
            if value is SimpleBasicType {
```

Comment (Nuvindu) [reply ballerina/utils.bal:77]: check for other places

File: ballerina/utils.bal [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+// under the License.
+
+import ballerina/url;
+
+type SimpleBasicType string|boolean|int|float|decimal;
+
+# Represents encoding mechanism details.
+type Encoding record {
+    # Defines how multiple values are delimited
+    string style = FORM;
```
Comment (Nuvindu) [inline ballerina/utils.bal:27]:
```suggestion
    EncodingStyle style = FORM;
```

File: ballerina/utils.bal [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+#
+# + value - Value to be encoded
+# + return - Encoded string
+isolated function getEncodedUri(anydata value) returns string {
+    string|error encoded = url:encode(value.toString(), "UTF8");
+    if (encoded is string) {
+        return encoded;
+    } else {
+        return value.toString();
+    }
```
Comment (Nuvindu) [inline ballerina/utils.bal:180]:
```suggestion
    if encoded is string {
        return encoded;
    }
    return value.toString();
```

File: ballerina/utils.bal (hunk 3 above)
Comment (ayeshLK) [reply ballerina/utils.bal:38]: These are generated code from Bal OpenAPI tool and hence this should be fixed in the tool itself. Currently there is an issue [1] to track this improvement in the tool. Hence, we will not fix this until the improvement is done for Bal OpenAPI tool.

[1] - [https://github.com/ballerina-platform/ballerina-library/issues/5812](https://github.com/ballerina-platform/ballerina-library/issues/5812)

File: ballerina/utils.bal (hunk 4 above)
Comment (ayeshLK) [reply ballerina/utils.bal:77]: These are generated code from Bal OpenAPI tool and hence this should be fixed in the tool itself. Currently there is an issue [1] to track this improvement in the tool. Hence, we will not fix this until the improvement is done for Bal OpenAPI tool.

[1] - [https://github.com/ballerina-platform/ballerina-library/issues/5812](https://github.com/ballerina-platform/ballerina-library/issues/5812)

File: ballerina/utils.bal (hunk 5 above)
Comment (ayeshLK) [reply ballerina/utils.bal:27]: These are generated code from Bal OpenAPI tool and hence this should be fixed in the tool itself. Currently there is an issue [1] to track this improvement in the tool. Hence, we will not fix this until the improvement is done for Bal OpenAPI tool.

[1] - [https://github.com/ballerina-platform/ballerina-library/issues/5812](https://github.com/ballerina-platform/ballerina-library/issues/5812)

File: ballerina/utils.bal (hunk 6 above)
Comment (ayeshLK) [reply ballerina/utils.bal:180]: These are generated code from Bal OpenAPI tool and hence this should be fixed in the tool itself. Currently there is an issue [1] to track this improvement in the tool. Hence, we will not fix this until the improvement is done for Bal OpenAPI tool.

[1] - [https://github.com/ballerina-platform/ballerina-library/issues/5812](https://github.com/ballerina-platform/ballerina-library/issues/5812)

**Review by Nuvindu — APPROVED** [review]
LGTM

**Pull request merged**